### PR TITLE
[DYN-3486] Extensions pin icon

### PR DESCRIPTION
**Pull request by Astul-Betizagasti**
### Purpose

Add a pin icon to the extensions menu button
image

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

Aaron Tang (@QilongTang )

### FYIs

Roberto Tellez (@RobertGlobant20 )

### CODE REVIEW

File: src/DynamoCoreWpf/UI/Themes/Modern/MenuStyleDictionary.xaml [hunk 1]
```diff
@@ -402,6 +402,98 @@
...
+                                          Margin="0,0,6,0"
+                                          VerticalAlignment="Center"
+                                          ContentSource="Icon" />
+                        <ContentPresenter Name="Content"
+                                          Margin="0"
+                                          Grid.Column="1"
+                                          VerticalAlignment="Center"
+                                          ContentSource="Header"
+                                          RecognizesAccessKey="True" />
+                        <Popup Name="Popup"
```
Comment (QilongTang) [inline src/DynamoCoreWpf/UI/Themes/Modern/MenuStyleDictionary.xaml:435]: What would this style refer to?

**Comment by QilongTang** [comment]
@Astul-Betizagasti Would you also attach the UI state where there are extensions sub menu items under it? I thought the diff here is the icon would be colorful when there are sub menu items, can you confirm?

File: src/DynamoCoreWpf/UI/Themes/Modern/MenuStyleDictionary.xaml (hunk 1 above)
Comment (Astul-Betizagasti) [reply src/DynamoCoreWpf/UI/Themes/Modern/MenuStyleDictionary.xaml:435]: As the changes for the icon only apply for the extensions menu I had to override the ControlTemplate for TopLevelHeader on it, and that ControlTemplate includes the Popup style configuration so I had to be duplicated. It is the same as the one on line 48

**Comment by Astul-Betizagasti** [comment]
> @Astul-Betizagasti Would you also attach the UI state where there are extensions sub menu items under it? I thought the diff here is the icon would be colorful when there are sub menu items, can you confirm?

The icon remains the same in any state, Sol mentioned yesterday on stand-up that for now we will use the grey color scheme as the colored one will be implemented in the future. I checked with him and Jingyi on today´s stand-up just to be sure and they confirmed that

**Comment by QilongTang** [comment]
> > @Astul-Betizagasti Would you also attach the UI state where there are extensions sub menu items under it? I thought the diff here is the icon would be colorful when there are sub menu items, can you confirm?
> 
> The icon remains the same in any state, Sol mentioned yesterday on stand-up that for now we will use the grey color scheme as the colored one will be implemented in the future. I checked with him and Jingyi on today´s stand-up just to be sure and they confirmed that

Thanks, Looks good to me. Please resolve the conflict and merge.

**Review by QilongTang — APPROVED** [review]

**Pull request merged**